src/app/app.component.js

```diff
@@ -30,15 +32,40 @@ function AppComponentCtrl(
 
     this.settings = dimSettingsService;
     $scope.$watch(() => this.settings.itemSize, (size) => {
-      document.querySelector('html').style.setProperty("--item-size", `${size}px`);
+      if (!this.isPhonePortrait) {
+        document.querySelector('html').style.setProperty("--item-size", `${size}px`);
+      }
     });
     $scope.$watch(() => this.settings.charCol, (cols) => {
-      document.querySelector('html').style.setProperty("--character-columns", cols);
+      if (!this.isPhonePortrait) {
+        document.querySelector('html').style.setProperty("--character-columns", cols);
+      }
     });
     $scope.$watch(() => this.settings.vaultMaxCol, (cols) => {
       document.querySelector('html').style.setProperty("--vault-max-columns", cols);
     });
 
+    const sizesMobile = {
+      3: 64,
+      4: 51,
+      5: 42
+    };
+    $scope.$watch(() => this.settings.charColMobile, (cols) => {
+      if (this.isPhonePortrait) {
+        document.querySelector('html').style.setProperty("--item-size", `${sizesMobile[cols]}px`);
+        document.querySelector('html').style.setProperty("--character-columns", cols);
+      }
+    });
+    $scope.$watch(() => this.isPhonePortrait, (isPhonePortrait) => {
```

Comment:
It's better to just handle this in the subscription to isPhonePortraitStream, than try to watch the result

Reply:
Now I'm lost...

Reply:
Oh, I forgot.  This may be misleading but the reason to also check for changes on isPhonePortrait is because when the mobile changes from portrait to landscape, since we don't refresh the page, we need to also change the settings responsible to the size/columns.
This check also works for when the user decreases the width of the window (on desktop) enough to trigger the isPhonePortrait.